addons/website_slides/models/slide_channel.py

```diff
@@ -625,21 +648,13 @@ def _get_earned_karma(self, partner_ids):
 
         return total_karma
 
-    def _remove_membership(self, partner_ids):
+    def _remove_membership(self, partner_ids, archive=False):
```

Comment:
I see that this method is also called when user fails his last item. In that case, the membership is not archived but deleted. Is it intended ? If I fail, do I need to restart from scratch ? Or can I keep the progression and karma gained with Quiz I already passed ?

Reply:
When the user fail his last attempt at a survey all his progress is deleted and he has to retake it completely from scratch. That's the way it was before this commit tough